app/src/main/java/com/nervousfish/nervousfish/modules/constants/Constants.java

```diff
@@ -6,22 +6,29 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.InvalidObjectException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+
 /**
  * Class implementing the main constants for the application.
  */
 public final class Constants implements IConstants {
 
+    private static final long serialVersionUID = -2465684567600871939L;
     private static final Logger LOGGER = LoggerFactory.getLogger("Constants");
     private final static String DB_USERDATA_PATH = "/accountInformation.json";
     private final static String DB_CONTACTS_PATH = "/contacts.json";
-    private final String androidFilesDir;
+    private final IServiceLocator serviceLocator;
+    private transient final String androidFilesDir;
 
     /**
      * Prevents construction from outside the class.
      *
      * @param serviceLocator Can be used to get access to other modules
      */
     private Constants(final IServiceLocator serviceLocator) {
+        this.serviceLocator = serviceLocator;
```

Comment:
That's right. However, the reference is just 8 bytes so practically nothing and just keeping the reference keeps the code consistent with the other modules and makes it easier in the future to use the service locator